### ISSUE TITLE
rundeck plugin error

### ISSUE THREAD

**Issue by gcianchet** [xebialabs/overthere#140]
Hello,

we are running the rundeck plugin to access remote machines via winrm.

Currently we have the following error message:

[overthere-winrm:172.27.0.214] failed: WinRM Error: Unexpected HTTP response on http://172.27.0.214:5985/wsman: (401)
Failed: WinRMProtocolError: WinRM Error: Unexpected HTTP response on http://172.27.0.214:5985/wsman: (401)
UBNAXMEN		Execution failed: 8: [Workflow step failures: {1=Dispatch failed on 1 nodes: [UBNAXUBTBP: WinRMProtocolError: WinRM Error: Unexpected HTTP response on http://172.27.0.214:5985/wsman: (401)]}, Node failures: {UBNAXUBTBP=[WinRMProtocolError: WinRM Error: Unexpected HTTP response on http://172.27.0.214:5985/wsman: (401)]}]

Can you please help?

**Comment by hierynomus** [comment]
Hi gcianchet,

Have you looked at our [WinRM Troubleshooting Guide](https://github.com/xebialabs/overthere#cifs_troubleshooting)? In there you will find most common error scenarios that occur when connecting with WinRM. Also have you correctly setup the remote machine using the [Host Setup guide](https://github.com/xebialabs/overthere#cifs_host_setup)?

**Comment by gcianchet** [comment]
Hello,

I have checked the troubleshooting guide and I examined the part called (I did it also before writing here):

"WinRM command fails with a 401 response code"

Here we have the following checks:

1. The Kerberos ticket is not accepted by the remote host
2. The WinRM service is not set up to accept unencrypted traffic.
3. The user is not allowed to log in.
4. The user is not allowed to perform a WinRM command

regarding points 2, 3, 4:

The winRM is setup to accept unencrypted traffic, here is the winrm config abstract:

Config
...
    Service
...
        AllowUnencrypted = true
        Auth
            Basic = true
            Kerberos = true
            Negotiate = true
            Certificate = false
            CredSSP = false
            CbtHardeningLevel = Relaxed

The user is a domain user that we use to log in and perform winrm commands and it has administrator rights.

Regarding the Kerberos ticket, in the spn list I can see that the wsman protocol is set for the remote machine.

What else should I look for?

**Comment by hierynomus** [comment]
Have you checked [this part](https://github.com/xebialabs/overthere#kerberos---source-host) of the documentation also? Especially the casing is important, domain names are always fully uppercase, and should be specified as such...

**Comment by gcianchet** [comment]
This is the spn output

CN=UBNAXUBTBP,OU=JustBuilt,DC=lm,DC=sambo-em,DC=com
	WSMAN/UBNAXUBTBP.lm.sambo-em.com
	WSMAN/UBNAXUBTBP

you mean that the domain here should be uppercase?

**Comment by gcianchet** [comment]
I have also set on the source host the krb5.ini, but I still have the same error.

**Comment by gcianchet** [comment]
something new.
the 401 issue was due to the fact that the username in the plugin was without domain.

I specified user@domain and the error 401, disappeared.

Now I have a different one: "cannot locate default realm". I am working on that, but I would appreciate suggestions.

**Comment by lmayorga1980** [comment]
@gcianchet same thing here `Cannot locate default realmRequest`...

I am using basic auth here but wondering if it supports LDAP users that are members of the Local Admin Group

**Comment by sky1979** [comment]
On my instance it worked, but i did't use kerberos, I kept getting the 401 error and this is what helped me:
- On the server the winrm setup i set this : Kerberos = false 
- I added the trusted host like this: winrm set winrm/config/client @{TrustedHosts="*"}.
- in the resources.xml in rundeck for my node i set up the username without the domain just like "rundeck" (i created that user like the userguide said) and the hostname in my case i set up the ip address

After i made those changes in my configuration it worked!
I hope it helps!
Salvador

**Comment by ssbarnea** [comment]
I am trying here to enable a node to run on Windows and I am faceing the same problem, the 401 response, even if we did follow the walkthroug. We do have a hard requirement for running it using a domain user (not a local machine user).

In order to simplify the already complex setup procedure, I picked the HTTP only approach. Here is theoutput of the currently configuration:
https://gist.github.com/ssbarnea/9ca38dfa527ddc2ccffa
